docs/source/ROL.rst

```diff
@@ -105,21 +104,42 @@ method.
     situations when the routine that evaluates the functional :bash:`J`
     fails. Such failures are usually due to failure in solving the state
     constraint. Among other reasons, this can happen when the control
-    :bash:`q` is not feasible (for instance, when the underlying mesh interstects itself)
-    or when the state constraint is nonlinear and the optimization step
-    is too large (in which case the initial guess is not good enough).
+    :bash:`q` is not feasible (for instance, when the underlying mesh
+    interstects itself) or when the state constraint is nonlinear and
+    the optimization step is too large (in which case the initial guess
+    is not good enough).
+
+    In a nutshell, trust-region methods solve a sequence of optimization
+    problems. In each of these, one minimizes a quadratic misfit functional
+    with control constraints. The idea is that the quadratic misfict functional
+    models the original misfit functional. The control constraint limits the
+    validity of this model to a trusted region. To construct the quadratic
+    misfit functional, one evaluates the original functional, its gradient,
+    and its Hessian (or a BFGS approximation of it) in a feasible point.
+    The minimizer to this quadratic functional is sought in a ball around
+    that feasible point (computing this minimizer is computationally inexpensive
+    and does not
+    involve further evaluations of the original functional or its derivatives).
+    Then, one evaluates the original misfit functional in this minimizer
+    and compares the *actual reduction* with the *predicted reduction*.
+    The new control is accepted if the actual reduction is
+    positive, that is, if there is actual reduction.
+    Then, if there is good agreement between the actual and predicted
+    reductions, the trust-region radius is increased. This radius is
+    decreased if the actual reduction is not positive or the ratio between
+    actual and predicted reductions is close to zero.
+    From this, we undestand that a safe solution to deal with failed
+    evaluations of :bash:`J` is to store the previously computed value of
+    :bash:`J` and, using a :bash:`try: ... except: ...` approach,
+    return it if the new evaluation of :bash:`J` fails. This corresponds
+    to a nonpositive actual reduction, which triggers a reduction of
+    the trust-region radius.
```

Comment:
Do we need to actually store the value or could we just return infinity? The latter would simply the code quite a bit